source/github-helpers/search-query.ts

```diff
@@ -84,56 +78,58 @@ export default class SearchQuery {
 		// Header nav example: is:open is:issue author:you archived:false
 		if (this.link.pathname === '/issues' || this.link.pathname === '/pulls') {
 			if (this.searchParams.has('user')) { // #1211
-				queries.push(`user:${this.searchParams.get('user')!}`);
+				queries.push('user:' + this.searchParams.get('user')!);
 			} else {
 				queries.push('author:@me');
 			}
 
 			queries.push('archived:false');
 		}
 
-		return queries.join(' ');
+		return queries;
 	}
 
-	getQueryParts(): string[] {
-		return splitQueryString(this.get());
+	get(): string {
+		return cleanQueryParts(this.queryParts).join(' ');
 	}
 
-	set(query: string): void {
-		const parts = splitQueryString(query);
-		const cleaned = cleanQueryParts(parts);
-		this.searchParams.set('q', cleaned.join(' '));
+	set(query: string): this {
+		this.queryParts = splitQueryString(query);
+		return this;
+	}
 
-		if (this.link && labelLinkRegex.test(this.link.pathname)) {
-			// Avoid a redirection to the conversation list that would erase the search query #5176
-			this.link.pathname = this.link.pathname.replace(/\/labels\/.+$/, '/issues');
+	applyChanges(): void {
+		this.searchParams.set('q', this.get());
+		if (this.link) {
+			this.link.search = String(this.searchParams);
+			if (labelLinkRegex.test(this.link.pathname)) {
+				// Avoid a redirection to the conversation list that would drop the search query #5176
+				this.link.pathname = this.link.pathname.replace(/\/labels\/.+$/, '/issues');
```

Comment:
Since `SearchQuery` doesn't change the ink automatically anymore, you can probably just move this line to the `set` method. This way no changes are made to the url until they're requested (and also since `.set` isn't automatically called in the constructor anymore)

Reply:
If we drop `this.link` then this will have to be extracted out of the class regardless.

Reply:
No, we still have to deal with the whole URLs inside, it's just that we don't store/expose the original link

Reply:
Firefox bug: https://github.com/refined-github/refined-github/pull/3208